compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <vector>
+
+namespace
+{
+
+class TaggedShapeAnalyzer final
+{
+public:
+  /**
+   * @brief check 'Transpose-Reshape-Transpose' can be replaced by one 'Reshape'.
+   *
+   * @example
+   *  Let's explain how analyzer check Transpose-Reshape-Transpose pattern with an exact example.
+   *
+   *  Let's assume under pattern is given :
+   *
+   *      Input(1, 7, 7, 448)
+   *            |
+   *      Transpose(perm=(0, 3, 1, 2))
+   *            |
+   *      Resahape(shape=(1, 448, 49))
+   *            |
+   *      Transpose(perm=(0, 2, 1))
+   *            |
+   *      Output(1, 49, 448)
+   *
+   *
+   *  It simulates how each dimension of the tensor's shape are transformed/moved
+   *  using a member variable named '_shape'.
+   *  'tags' in _shape record the initial order of each dimension.
+   *
+   *   TIMELINE              |   _shape states :
+   *                         |
+   *   init_shape_with_tag   |    _shape :
+   *                         |        - value :   (1)   (7)   (7)   (448)
+   *                         |        - tags  :   (-)   (0)   (1)   (2)
+   *                         |
+   *   analyze_transpose     |    _shape :
+   *                         |        - value :   (1)   (448)  (7)  (7)
```

Comment:
I updated a bit to remove some empty lines.

Personally, I prefer some space(empty lines) to discriminate the state easily. 
So, Rather than remove every empty line, I left some.